oktoberfest/predict/predictor.py

```diff
@@ -0,0 +1,197 @@
+from __future__ import annotations
+
+import importlib.util
+import logging
+from pathlib import Path
+from typing import Dict, List
+
+import anndata
+import numpy as np
+import pandas as pd
+
+from ..data.spectra import FragmentType, Spectra
+from ..utils import Config, group_iterator
+from .alignment import _alignment, _prepare_alignment_df
+from .dlomix import DLomix
+from .koina import Koina
+
+logger = logging.getLogger(__name__)
+
+
+class Predictor:
+    """Abstracts common prediction operations away from their actual implementation via the DLomix or Koina interface."""
+
+    def __init__(self, predictor: Koina | DLomix):
+        """Initialize from Koina or DLomix instance."""
+        self._predictor = predictor
+
+    @classmethod
+    def from_config(cls, config: Config, model_type: str, **kwargs) -> Predictor:
+        """Load from config object."""
+        model_name = config.models[model_type]
+
+        # TODO add documentation for config
+        if model_type == "irt" or not config.predict_intensity_locally:
+            logger.info(f"Using model {model_name} via Koina")
+            koina = Koina(model_name=model_name, server_url=config.prediction_server, ssl=config.ssl, **kwargs)
+            return Predictor(koina)
+
+        DLomix.initialize_tensorflow()
+
+        if not importlib.util.find_spec("dlomix"):
+            logger.exception(
+                ModuleNotFoundError(
+                    """Local prediction configured, but the DLomix package could not be found. Please verify that the
+                     optional DLomix dependency has been installed."""
+                )
+            )
+
+        output_folder = config.output / "data/dlomix"
+
+        if model_name == "baseline":
+            return Predictor(DLomix(model_type, model_name, output_folder))
+
+        model_path = Path(model_name)
+        if model_path.exists():
+            logger.info(f"Loading pre-trained PrositIntensityPredictor from {model_path}")
+            return Predictor(DLomix(model_type, model_path, output_folder))
+        else:
+            logger.critical(f"Specified model path {model_name} does not exist, please check")
+            logger.exception(FileNotFoundError())
+
+        return
+
+    def predict_intensities(self, data: anndata.AnnData, chunk_idx: List[pd.Index] | None = None, **kwargs):
```

Comment:
this should be a spectra object, please always use that, since it makes a difference in typing (see implementation in `Spectra.from_hdf5`. If type checking fails here and that is the reason you annotated as `AnnData`, sth. is wrong with the provided spectra object.